### DISCUSSION TITLE
Creating Snap7 Server on Linux?

### DISCUSSION THREAD

**Discussion by jwhitmor** [gijzelaerr/python-snap7#478] [Q&A] [answered]
I wanted to create a dummy Snap7 server on a Linux machine to test a snap7 client. Executing the documented server from the command line "python -m snap7.server --port 102" failed with a runtime warning: 

    /usr/lib/python3.8/runpy.py:127: RuntimeWarning: 'snap7.server' found in sys.modules after import of package 'snap7', but prior to execution of 'snap7.server'; this may result in unpredictable behaviour

I then decided I'd create a class based on the Server class, but that complained as well. So I looked at the Server class code and it looks like it's looking for a snap7 DLL which is Windows only, I assume? So is the Snap7 Server only available on Windows?

**Comment by spreeker** [comment]
No most definatly not Windows only. It was designed / tested in a linux environment.  See code below.

https://github.com/gijzelaerr/python-snap7/blob/master/snap7/common.py#L10

**Comment by gijzelaerr** [comment, marked as answer]
for me this works:
```
$ python3 -m snap7.server --port 102
[INFO]: creating server
[INFO]: setting event callback
[INFO]: registering area 5, index 1, size 100
[INFO]: registering area 1, index 1, size 100
[INFO]: registering area 4, index 1, size 100
[INFO]: registering area 3, index 1, size 100
[INFO]: registering area 5, index 0, size 976
[INFO]: starting server on 0.0.0.0:102
[INFO]: callback event: 2023-11-27 19:01:05 Server started
[INFO]: callback event: 2023-11-27 19:01:05 Server started
[INFO]: 2023-11-27 19:01:05 Server started
```
I think this might be related to the `runpy.py` script you have as an alias for your python interpreter, you have a manual `PYTHONPATH` environment variable that causes issues or you are running your code from a subdirectory of the python-snap7 library.

https://docs.python.org/3/library/runpy.html

what distribution and python version are you using and how did you install snap7?
